### ISSUE TITLE
CollectionInterface has no method registerRollback()

### ISSUE THREAD

**Issue by Sweetchuck** [consolidation/Robo#465]
According to PhpDoc the
`\Robo\Collection\Collection::$parentCollection` is an object which implements the `\Robo\Collection\CollectionInterface` but it has not `::registerRollback()` method, so the `\Robo\Collection\Collection::registerRollback()` can't call it.

https://github.com/consolidation/Robo/blob/1908cec67c6278d6d21535f14ca403cc860d4518/src/Collection/Collection.php#L344

In the real world the `\Robo\Collection\Collection::$parentCollection` is an `\Robo\Collection\Collection` instance. so it has `::registerRollback()` method, but something is inconsistent.

**Comment by greg-1-anderson** [comment]
Yes; CollectionInterface is mostly-complete, but not completely complete. This area of the code could use some work.  Collection could use some simplification.
